Add library conventions section (#18)
LibrePCB/librepcb-doc: library_conventions/symbols.adoc

So you should name it something like "Diode" and use the same symbol also
for all other standard diodes. Of course every kind of diode (e.g. Zener)
should have its own symbol because they look different.

On the other side, there are many very specific components, for example
a microcontroller. Even if it's possible to also use generic symbols for
them (e.g. "32-Pin IC"), you should create a symbol specific for that part
instead. This way you can choose a reasonable
<<libraryconventions-symbols-pinplacement,pin placement>>.

[#libraryconventions-symbols-naming]
==== Naming

Following conventions apply to symbol names:

- Language must be American English (en_US)
- Title case (e.g. "Capacitor Bipolar" instead of "Capacitor bipolar")
- Singular names, not plural (e.g. "Diode" instead of "Diodes")
- If reasonable, start with the generic term (e.g. "Supply GND" instead
  of "GND Supply") to improve navigation in sorted lists (all supply
  symbols are listed next to each other)

// Image spans across multiple sections
image::img/symbol_stm32.png[alt="Example Symbol",role="right"]

[#libraryconventions-symbols-origin]
==== Origin

> The origin _(0, 0)_ must be in the center of the symbol (not inclusing

Author: ```suggestion
The origin _(0, 0)_ must be at the center of the symbol (not including
```

> text elements). For non-symmetrical symbols it must be as close as

Author: ```suggestion
text elements). For non-symmetrical symbols it should be as close as
```

[#libraryconventions-symbols-pinplacement]
==== Pin Placement

- For integrated circuit symbols (i.e. rectangular outline), generally
  *don't place pins at the top and bottom edges*, but only on the left and
  the right. This helps to get clear, easily readable schematics.
- *Group pins by functionality*, not by physical location of the leads or by
  datasheet. Always keep the typical application circuit in mind and choose
  pin locations which help to get clear schematics with only few crossed-over
  net lines. For example put _GND_ exactly 5.08mm below the _VCC_ pin if it's
  likely that capacitors need to be connected to them (capacitors have a
  height of 5.08mm). Or place _D+_ and _D-_ of a USB device right on top of
  each other (with the default distance of 2.54mm) as they are always used
  as a pair.

[#libraryconventions-symbols-pinnaming]
==== Pin Naming

If the function of a pin is absolutely clear (e.g. anode/cathode of a diode),
choose its abbreviated functionality as name (e.g. "A" for anode and "C" for
cathode). If the functionality is not clear in the symbol (because it's
defined by the component using that symbol), just use numbers starting with

> "1" at top left and incrementing counterclockwise.

Author: I think this would be better:

```suggestion
"1" at top left and increment them counterclockwise.
```

Although both versions are still a bit hard to read.

[#libraryconventions-symbols-textelements]
==== Text Elements

Typical symbols should have exactly two text elements: `{{NAME}}` and
`{{VALUE}}`.

image::img/symbol_text_anchor_ic.png[alt="IC Symbol Text Anchor",role="right"]

For rectangular symbols, the name should be placed at top left, aligned at
bottom left to the corner of the symbol outlines. And the value should be
placed at bottom left, aligned at the top left to the corner of the symbol
outlines.

image::img/symbol_text_anchor_crystal.png[alt="Crystal Symbol Text Anchor",role="left"]

> Other shaped symbols may have text elements placed differently, see for

Author: ```suggestion
Irregularly shaped symbols may have text elements placed differently, see for
```

> example this crystal. Keep in mind that the value of a component can consist

Author: Which crystal? :slightly_smiling_face:

Reviewer: Changed to "the crystal at the left".

Author: Ah sorry, I did not realize that there was an image above the paragraph, I only looked for one after it :confused:

Reviewer: Because of that I posted the link to the rendered output, that's much easier to review because you see the layout ;)

[#libraryconventions-symbols]
=== Symbol Conventions

[NOTE]
====
These guidelines are not yet complete. Help us create sensible conventions
https://github.com/LibrePCB/librepcb-doc/issues?q=is%3Aissue+label%3A%22Conventions%3A+Symbols%22[on GitHub]!

> ====

Author: I'd add the note to every library element documentation, with the appropriate Github label preselected. That makes it a bit easier to find issues :)

==== Pin Placement

Preferrably, don't place pins at the top or bottom. Place them on the left or

> right side of a symbol instead.

Reviewer 2: > What I meant is that "IC-like" symbols (big yellow block with many pins) should not have pins at top/bottom.

What's the reason for this? I have seen IC-like symbols with pins at top/bottom (GND and Vcc for example) and didn't have any problem with that.

> right side of a symbol instead.

Reviewer: IMHO schematics are more readable if symbols with many pins have pins only at left and right, especially those pins which require capacitors close to them.

Example (see microcontroller):

![all_sides](https://i.stack.imgur.com/k3Gbj.png)

vs

![left_right](https://www.proggen.org/lib/exe/fetch.php?media=electronics:stm32:phy-schematic.png)
